### PR TITLE
MRG: use mne resample function

### PR DESCRIPTION
**Pull request by rkmaddox**
Addresses #248. Only difference is the mne `resample` has a `verbose` argument. Should I set that to `False` so it behaves as before, or not worry about it?

### CODE REVIEW

**Comment by rkmaddox** [comment]
@drammock, in case you're not subscribe.

**Comment by rkmaddox** [comment]
`make flake` causing tests to fail.

**Comment by rkmaddox** [comment]
Correction: 
Lacking `future` module causing appveyor to fail.
`make flake` causing Travis to fail.

**Comment by Eric89GXL** [comment]
WTF Pyglet

Can you change the CIs to `pip install pyglet` instead of installing the `tip.zip`? We shouldn't need the tip anymore.

File: expyfun/stimuli/_filter.py [hunk 1]
```diff
@@ -7,118 +7,11 @@
...
-
-    # Restore the original array shape (modified for resampling)
-    y.shape = orig_shape[:-1] + (y.shape[1],)
-    if axis != orig_last_axis:
-        y = y.swapaxes(axis, orig_last_axis)
-
-    return y
-
-
 def _fft_resample(x, W, new_len, npad, to_remove):
```
Comment (Eric89GXL) [inline expyfun/stimuli/_filter.py]: we shouldn't need this, either

**Comment by rkmaddox** [comment]
Changed the `pip` line and added `mne` as a dependency, but seems like `make flake` still causes a problem in Travis. Any thoughts?

**Comment by rkmaddox** [comment]
Appveyor now happy. Travis still sad for aforementioned reason.

**Comment by Eric89GXL** [comment]
You can combine those two pip install lines into one command, "pip install mne pyglet"

Do we still need _smart_pad? git grep it. Can you also check the other functions in that file? Not sure how many were related only to resampling. 

Get rid of unused imports to make Travis happy. You might want to enable pyflakes in spyder so you can see those while editing.

**Comment by Eric89GXL** [comment]
You can also run make flake locally so you don't have to wait for Travis for feedback

**Comment by rkmaddox** [comment]
Ah, I understand. I thought it was a problem with pyflake, not that it was catching problems with our code. Should be fixed now.

**Comment by Eric89GXL** [comment]
Check coverage CI, `_smart_pad` isn't needed anymore.

Can you update the README.rst to say mne is required now? Although maybe we should actually only need it for stimulus generation, not e.g. presentation. So in that case, what we need to do is something like:

```
try:
    from mne.filter import resample as _resample
except ImportError:
    _resample = None


def resample(...)
    """..."""
    if _resample is None:
        raise ImportError(...)
    _resample(...)
```

**Comment by Eric89GXL** [comment]
or you can nest the `mne` import within the `resample` function itself if you want

**Comment by rkmaddox** [comment]
There is no `resample` function. It just imports the MNE one. However, I was thinking about changing that. `mne`'s `resample` assumes that outside of the signal are mirror-imaged versions of that signal, but it is more typical to assume zeros, and that is what the previous function did.

So I was considering making `expyfun`'s `resample` wrap `mne`'s. Seem like an OK plan?

**Comment by rkmaddox** [comment]
Ohhhhhhhhh I just understood how the `resample` function was being created. Yikes. I didn't really process that code at the bottom.

**Comment by Eric89GXL** [comment]
So you know what to do, or is it still unclear?

**Comment by rkmaddox** [comment]
Is this sufficient?

```python
"""Stimulus resampling functions
"""

from distutils.version import LooseVersion

import mne
if LooseVersion(mne.__version__) < LooseVersion('0.8'):
    raise ImportError('mne-python too old')
from mne.filter import resample  # NOQA
```

**Comment by rkmaddox** [comment]
That doesn't take care of the edge handling, though. Just not sure if it _should_ be taken care of. (I think it probably should, as this resample function will be used primarily for audio, rather than brain signals.)

File: .travis.yml [hunk 2]
```diff
@@ -34,12 +34,8 @@ before_install:
...
     # Only install these deps if necessary (slow on Py3k)
     - travis_retry pip install -q nose-timer flake8;
-    - if [ "${PYTHON}" == "3.4" ]; then
-        travis_retry pip install -q pyglet;
-      else
-        travis_retry pip install -q http://bitbucket.org/pyglet/pyglet/get/tip.zip;
-      fi;
-    # Full dependencies (pandas, mne-python, h5py, joblib)
+    - travis_retry pip install -q pyglet mne;  # mne is a dependency now that resample is taken from it
+    # Full dependencies (pandas, mne-python, h5py, joblib, mne-python)
```
Comment (drammock) [inline .travis.yml]: since you're mucking around in this file, note the duplicated "mne-python" in this comment

**Comment by rkmaddox** [comment]
Never mind. It appears it pads with zeros, and I was remembering incorrectly. This should be good to go, I think, if I make those changes from the previous comment.

**Comment by Eric89GXL** [comment]
I don't think `mne` should be a mandatory requirement for `import expyfun`, but it can be a mandatory requirement for `from expyfun.stimuli import resample; resample(...)`, make sense? See my comment above for suggested mechanism.

**Comment by rkmaddox** [comment]
Ah, yes. I understand now.

**Comment by rkmaddox** [comment]
Is the call to LooseVersion still necessary, or will it also cause an import error if someone tries to import `resample` from `mne.filter` in an older install?

**Comment by Eric89GXL** [comment]
I wouldn't worry about it in this case, 0.8 is over a year old now and I doubt any of our user base is using an old version. But yeah, I suspect that a pre-0.8 will either give an `ImportError` or maybe some other error when you go to actually use it, which is fine.

**Comment by rkmaddox** [comment]
I think this should be good to go, but take a look at the docstring thinger. I wanted it to stay current with the function, but not sure if what I did was kosher.

**Comment by rkmaddox** [comment]
Never mind. The way I did it it doesn't show the arguments. I'll just do it the less elegant but probably more functional way.

**Comment by Eric89GXL** [comment]
Yeah, unfortunately the `args/kwargs` + docstring is something that I haven't found a way around violating DRY to get right in these situations

**Comment by rkmaddox** [comment]
The `__doc__` trick worked OK. But I did not see a way to get the args to work.

**Comment by drammock** [comment]
LMK when this is ready for an extra set of eyes and a `make test` on my machine

**Comment by rkmaddox** [comment]
Hmm. How about this? Seems to work on my machine.

```python
def _resample_error(*args, **kwargs):
    """mne-python is required to use the resample function
    """
    raise ImportError('mne-python is required to use the resample function')

try:
    from mne.filter import resample
except ImportError:
    resample = _resample_error
```

**Comment by rkmaddox** [comment]
@drammock ready for test. Changing to MRG.

**Comment by Eric89GXL** [comment]
LGTM, +1 for merge. @rkmaddox can you look quickly at why our coverage went down? Maybe we just removed a well-covered file, which is what I suspect happened.

**Comment by rkmaddox** [comment]
Yep, it's because the number of covered lines in `_filter.py` went down by ~60.

**Comment by rkmaddox** [comment]
That's what we get for DRY.

**Comment by drammock** [comment]
weird, the final commit 57363048775ce862065c3d67763f6c9756d9abb6 passed all tests, but the homepage is showing [appveyor failure](https://ci.appveyor.com/project/Eric89GXL/expyfun/branch/master), failing on "Testing EC with TDT" line.

**Comment by Eric89GXL** [comment]
I set it to rebuild, might have just been a fluke

**Comment by Eric89GXL** [comment]
all better

**Comment by drammock** [comment]
thanks @Eric89GXL

**Pull request merged**